### ISSUE TITLE
Watchdog for RSemaphore

### ISSUE THREAD

**Issue by abagiev** [redisson/redisson#2933]
RLock is great mechanism for distributed locking since it has a watchdog for releasing lock when client application is dead. 

However, when it comes to reactive version (RLockReactive) it becomes totally unuseful because unlocking can be done only by the thread that performs locking. And with reactive streams this never happens for sure.

Instead, it is suggested to use RSemaphore as an alternative since it is not bound to threads. In general, it does the job. However, when client application becomes dead, already performed lock hangs forever. Setting key expiration time partially solves the problem, however, actual atomic operation time may differ, so it is usually difficult to choose the correct timeout.

Another option is a watchdog, but apparently Redisson one does not work with RSemaphore.

So, is it possible to configure the Redisson watchdog for prolonging RSemaphore expire time? Or the only way out is implementing own watchdog?

**Comment by mrniko** [comment]
I can suggest you to use `RPermitExpirableSemaphore` so you can define `leaseTime` per permit.

**Comment by tomasz-galuszka** [comment]
I have a case when I used the `RPermitExpirableSemaphore` as a lock for `RMapCacheReactive` **get by key** method.
The flow:
1) I get the element by key
2) If not exists I create a semaphore which includes a **map key in the name** then I do `trySetPermits(1)` and `tryAcquire(lockWaitTime.toMillis(), lockLeaseTime.toMillis(), MILLISECONDS)`
3) Then I do `flatMap` and run a function to get the value with attached `doFinally` which calls `semaphore.release(permitId)`

All works fine for 50, 100, 200, 500 asynchronous executions (the function to get the data is executed only once), great !

My problem is when I checked the redis KEYS. There are multiple KEYS related to my semaphore name and they stays there forever. Is there an easy way to clean them up when not used for some time ? 

Or do I need to do something like below by myself ? 
````
                mySemaphore.expire(lockExpireTime.toMillis(), MILLISECONDS)
                        .doOnNext(r -> log.info("Set expired: {}", r)).subscribe();
````
